### PR TITLE
Add mx.array.__format__

### LINKED ISSUE #1520: [FEATURE] Adding (string) __format__ to values extracted from array
**Describe the feature**
Be able to use f-strings formatting on elements of arrays.

**To Reproduce**
```python
import mlx.core as mx
a = mx.arange(1)
print(f"{a[0]:.3f}")
# TypeError: unsupported format string passed to array.__format__
b = mx.random.normal()
print(f"{b:.3f}")
# TypeError: unsupported format string passed to array.__format__
```

**Expected behavior**
Displaying the float to the nth digit like `print(f"{float(a[0]):.{n}f}")` would do, without converting to a float.

**Desktop:**
 - MacBook Pro M3 Max (14 cores, 96 GB)
 - MacOS 14.6.1
 - MLX 0.19.0

**Additional context**
I think it would help make the transition from NumPy or PyTorch slightly smoother.

### PR DESCRIPTION
**Pull request by barronalex**
Resolves #1520

### CODE REVIEW

File: python/tests/test_array.py [hunk 1]
```diff
@@ -1875,6 +1875,17 @@ def test_to_scalar(self):
         with self.assertRaises(ValueError):
             int(a)
 
+    def test_format(self):
+        a = mx.arange(3)
+        self.assertEqual(f"{a[0]:.2f}", "0.00")
+
+        b = 0.35487
+
+        self.assertEqual(f"{b:.1f}", "0.4")
```
Comment (awni) [inline python/tests/test_array.py:1884]: I don't think this is testing anything in MLX?

File: python/src/array.cpp [hunk 2]
```diff
@@ -845,6 +845,16 @@ void init_array(nb::module_& m) {
           nb::rv_policy::none)
       .def("__int__", [](array& a) { return nb::int_(to_scalar(a)); })
       .def("__float__", [](array& a) { return nb::float_(to_scalar(a)); })
+      .def(
+          "__format__",
+          [](array& a, nb::object format_spec) {
+            if (a.size() > 1) {
```
Comment (awni) [inline python/src/array.cpp:851]: I notice numpy only supports scalar (`a.ndim() == 0`). Should we do the same?

**Review by awni — APPROVED** [review]
Thanks for the fast fix!! Left a couple minor comments. Take a look then feel free to merge when ready.

File: python/tests/test_array.py (hunk 1 above)
Comment (barronalex) [reply python/tests/test_array.py]: That's silly -- good catch!

**Pull request merged**